sip-test-kit/src/test/java/de/ikor/sip/foundation/testkit/config/CamelContextLifecycleHandlerTest.java

```diff
@@ -0,0 +1,77 @@
+package de.ikor.sip.foundation.testkit.config;
+
+import static org.mockito.Mockito.*;
+
+import de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.RouteInvoker;
+import de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.impl.DefaultRouteInvoker;
+import de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.impl.FileRouteInvoker;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.camel.Endpoint;
+import org.apache.camel.ExtendedCamelContext;
+import org.apache.camel.Route;
+import org.apache.camel.component.file.FileEndpoint;
+import org.apache.camel.impl.engine.DefaultRouteController;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class CamelContextLifecycleHandlerTest {
```

Comment:
Would it be possible to test this behavior for real? - Starting context instead of mocking it

Reply:
Invested some time to find proper way for testing, but couldn't find anything useful.
Covered all possible cases with unit tests.